### PR TITLE
Enabling to configure renew refresh token property per application

### PR DESCRIPTION
**Pull request by Dumindu-Kanchana**
Improvement for: wso2/product-is#4550

### Proposed changes in this pull request

This feature will enable to configure renew refresh token property value per application.

### CODE REVIEW

**Comment by nilasini** [comment]
@emswbandara @JKAUSHALYA

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/dao/OAuthAppDO.java [hunk 1]
```diff
@@ -283,4 +284,11 @@ public boolean isBypassClientCredentials() {
     public void setBypassClientCredentials(boolean isPublicClient) {
         this.bypassClientCredentials = isPublicClient;
     }
+
+    public void setRenewRefreshTokenEnabled(String renewRefreshTokenEnabled){
```
Comment (Dumindu-Kanchana) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/dao/OAuthAppDO.java]: formatting Issue

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/dao/OAuthAppDO.java [hunk 2]
```diff
@@ -283,4 +284,11 @@ public boolean isBypassClientCredentials() {
...
+
+        this.renewRefreshTokenEnabled = renewRefreshTokenEnabled;
+    }
+
+    public String getRenewRefreshTokenEnabled() { return renewRefreshTokenEnabled; }
```
Comment (Dumindu-Kanchana) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/dao/OAuthAppDO.java]: add 2 new lines

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/RefreshGrantHandler.java [hunk 3]
```diff
@@ -574,4 +574,20 @@ private static void addUserAttributesToCache(AccessTokenDO accessTokenBean,
             AuthorizationGrantCache.getInstance().addToCacheByToken(authorizationGrantCacheKey, grantCacheEntry);
         }
     }
+
+    private boolean getRenewRefreshTokenProperty(String renewRefreshToken){
```
Comment (Dumindu-Kanchana) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/RefreshGrantHandler.java]: formatting issue

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/RefreshGrantHandler.java [hunk 4]
```diff
@@ -574,4 +574,20 @@ private static void addUserAttributesToCache(AccessTokenDO accessTokenBean,
...
+
+        if (renewRefreshToken != null){
```
Comment (Dumindu-Kanchana) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/RefreshGrantHandler.java]: formatting issue

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/RefreshGrantHandler.java [hunk 5]
```diff
@@ -574,4 +574,20 @@ private static void addUserAttributesToCache(AccessTokenDO accessTokenBean,
...
+            if (log.isDebugEnabled()) {
+                log.debug("Reading the Oauth application specific renew " +
+                        "refresh token value as " + renewRefreshToken + " from the IDN_OIDC_PROPERTY table");
+            }
+            return Boolean.parseBoolean(renewRefreshToken);
+        } else {
+            if (log.isDebugEnabled()) {
+                log.debug("Rading the global renew refresh token value from the identity.xml");
```
Comment (Dumindu-Kanchana) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/RefreshGrantHandler.java]: spelling

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/RefreshGrantHandler.java [hunk 6]
```diff
@@ -574,4 +574,20 @@ private static void addUserAttributesToCache(AccessTokenDO accessTokenBean,
             AuthorizationGrantCache.getInstance().addToCacheByToken(authorizationGrantCacheKey, grantCacheEntry);
         }
     }
+
+    private boolean getRenewRefreshTokenProperty(String renewRefreshToken) {
+
+        if (renewRefreshToken != null) {
```
Comment (emswbandara) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/RefreshGrantHandler.java]: shall we use StringUtils.isNotBlank here?

Comment (Dumindu-Kanchana) [reply components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/RefreshGrantHandler.java]: fixed with 6c8dd0427e303578e81605a727761d4d05981af5

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/RefreshGrantHandler.java [hunk 7]
```diff
@@ -574,4 +574,20 @@ private static void addUserAttributesToCache(AccessTokenDO accessTokenBean,
             AuthorizationGrantCache.getInstance().addToCacheByToken(authorizationGrantCacheKey, grantCacheEntry);
         }
     }
+
+    private boolean getRenewRefreshTokenProperty(String renewRefreshToken) {
```
Comment (ruwanta) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/RefreshGrantHandler.java]: we need to change the method name to "isRenewRefreshToken(..)" and use it like
if(! isRenewRefreshToken(..) {
..
} else {
}

As it is more readable

Comment (Dumindu-Kanchana) [reply components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/RefreshGrantHandler.java]: fixed with 0d23b1b5c857ce0897ca3526455f2de3fc55784e

**Pull request merged**